crates/aiken-lang/src/gen_uplc.rs

```diff
@@ -254,7 +255,7 @@ impl<'a> CodeGenerator<'a> {
                 panic!("Dangling expressions without an assignment")
             };
 
-            let replaced_type = convert_opaque_type(tipo, &self.data_types);
+            let replaced_type = convert_opaque_type(tipo, &self.data_types, true);
```

Comment:
Note: the flag now indicates whether we want a deep (true) or shallow (false) conversion. This is necessary because as we reify types into untyped expression, we need to preserve **nested** replaceable opaque types until we actually meet them. In code-gen however, we go brrrrrrr.

Reply:
Codegen does go brrrrrrr